### PR TITLE
feat: Ensure overwritten operations check for sufficient gas first

### LINKED ISSUE #11377: Improve gas checks for overwritten operations
### Problem

Checks for sufficient gas should occur at the beginning of all overwritten operations in order to ensure that needless work is not performed.

### Solution

Move checks for sufficient gas to the beginning of all overwritten evm operations. 

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by lukelee-sl**
**Description**:
Make sure that all overwritten operations check for gas before performing any other tasks.  

These checks are largely redundant as they are also performed in the base operations in Besu but these checks were added at the beginning of each operation execution to ensure that there is enough gas before doing _minimal_ work not related to determining the gas required to perform the operation.

**Related issue(s)**:

Fixes #11377 

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/operations/BasicCustomCallOperation.java [hunk 1]
```diff
@@ -91,6 +93,11 @@ default Operation.OperationResult executeChecked(@NonNull final MessageFrame fra
         requireNonNull(evm);
         requireNonNull(frame);
         try {
+            final long cost = cost(frame);
```
Comment (david-bakin-sl) [inline hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/operations/BasicCustomCallOperation.java:96]: (Sigh.  This `cost()` takes a `Frame` whereas the original Besu `cost()` takes a boolean.  Especially noticeable in this particular PR where _both_ are used! Can't change the Besu one, so wish this one had a different name ... darn.)

File: hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/operations/CustomBalanceOperation.java [hunk 2]
```diff
@@ -47,8 +48,12 @@ public CustomBalanceOperation(
     @Override
     public OperationResult execute(@NonNull final MessageFrame frame, @NonNull final EVM evm) {
         try {
-            final var address = Words.toAddress(frame.getStackItem(0));
+            final long cost = cost(false);
```
Comment (david-bakin-sl) [inline hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/operations/CustomBalanceOperation.java:51]: Is it ok that we want sufficient gas for a _cold_ operation but we charge (L64 & L68) for the (less expensive?) _warm_ operation?

(Same question for some of the other classes too.)

**Review by david-bakin-sl — COMMENTED** [review]
I have a question about warm vs cold gas.  So ... this is a comment.  Ping me!

File: hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/operations/CustomBalanceOperation.java (hunk 2 above)
Comment (lukelee-sl) [reply hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/operations/CustomBalanceOperation.java:51]: Good point.  I think the right thing to do is to charge for cold operation as it is unlikely that the address has been warmed by `super.execute()`

**Review by david-bakin-sl — APPROVED** [review]
LGTM

**Pull request merged**